Source/WebCore/page/EventHandler.cpp

```diff
@@ -3879,28 +3879,6 @@ bool EventHandler::needsKeyboardEventDisambiguationQuirks() const
 }
 #endif
 
-#if ENABLE(FULLSCREEN_API)
-bool EventHandler::isKeyEventAllowedInFullScreen(const PlatformKeyboardEvent& keyEvent) const
-{
-    RefPtr document = m_frame->document();
-    if (document->fullscreen().isFullscreenKeyboardInputAllowed())
```

Comment:
`isFullscreenKeyboardInputAllowed()` is defined as `return fullscreenElement() && m_areKeysEnabledInFullscreen;`. It looks like `m_areKeysEnabledInFullscreen` gets set under certain circumstances where the ChromeClient wishes to reject keyboard input.

I wonder if we should revise this method to just be:
```
#if ENABLE(FULLSCREEN_API)
bool EventHandler::isKeyEventAllowedInFullScreen(const PlatformKeyboardEvent& keyEvent) const
{
    if (RefPtr document = m_frame->document()) {
        return document->fullscreen().isFullscreenKeyboardInputAllowed())

    return true;
}
```

Reply:
@nt1m Can you remind us of why `DocumentFullscreen` has cases where `m_areKeysEnabledInFullscreen` might be set to false? I think this might just be set based on whether the Document fullscreen feature is enabled or not, so we probably don't want to lose this behavior ( unless that feature has been removed).

Reply:
I'm pretty sure it's just iOS's spoofing restriction in general that disallows using the keyboard in fullscreen:
https://searchfox.org/wubkat/rev/7cd5229e664679bd5aa646fce5a6f6e9ae77aa20/Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp#176-186

@jernoble can say more, but it's to avoid webpages from being able to emulate the system UI in fullscreen and stealing data, so anything that remotely looks like typing will be disallowed.

Reply:
The only other check I see is `isDocumentFullscreenEnabled()` which we already check for in `DocumentFullscreen::requestFullscreen()` earlier on. If security is OK with iPhone fullscreen also allowing keystrokes, then Brandon's patch is fine (although there would be more to clean up). If iPhone should behave differently from iPad and preserve current behavior, then this patch should be revised

Reply:
But I think you are wrong, @nt1m. It looks like in the old code we checked `DocumentFullScreen:: isFullscreenKeyboardInputAllowed()` which is controlled by various feature flag settings. I think we should retain this, in case a WebKit embedder wants to deny keyboard access for some reason (or else we should remove those feature flags!)

Reply:
This is the only place that sets the flag you're mentioning: https://searchfox.org/wubkat/rev/7cd5229e664679bd5aa646fce5a6f6e9ae77aa20/Source/WebCore/dom/DocumentFullscreen.cpp#224-233
https://searchfox.org/wubkat/rev/7cd5229e664679bd5aa646fce5a6f6e9ae77aa20/Source/WebCore/dom/DocumentFullscreen.cpp#282

This is the check that those call sites refer to, which I do not see any API hooks for:
https://searchfox.org/wubkat/rev/7cd5229e664679bd5aa646fce5a6f6e9ae77aa20/Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp#176-186